packages/pnpm-sync-lib/etc/pnpm-sync-lib.api.md

```diff
@@ -39,6 +51,11 @@ export interface IPnpmSyncCopyOptions {
     // (undocumented)
     getPackageIncludedFiles: (packagePath: string) => Promise<string[]>;
     // (undocumented)
+    logMessageCallback: (message: string, messageKind: ILogMessageKind, details?: {

```

Comment:
I suggest to move these parameters into an `ISyncLogMessageCallbackOptions` and `IPrepareLogMessageCallbackOptions` interface, so that in the future if needed, we could can add more parameters without breaking the API contract.  It will also make the caller's code more readable.

Compare this:

```ts
pnpmSyncPrepare({
  logMessageCallback: (m, k, d) => {
    // What is m? what is d??
    if (k === 'warning') { console.log(m); }
  }
});
```

or even worse...
```ts
pnpmSyncPrepare({
  logMessageCallback: (messageKind, message, details) => {
    // Why is this code malfunctioning? It looks entirely correct!
    if (messageKind === 'warning') { console.log(message); }
  }
});
```